odoo/tools/misc.py

```diff
@@ -60,6 +60,8 @@
 # Configure default global parser
 etree.set_default_parser(etree.XMLParser(resolve_entities=False))
 
+NON_BREAKING_SPACE = " "
```

Comment:
Have a look at how `format_amount` in this same file is implemented

Maybe `formatLang` or `format_amount` can be deprecated and using the other one.

Reply:
I've changed the definition:

```py
NON_BREAKING_SPACE = u'\N{NO-BREAK SPACE}'
```

As for `formatLang` vs `format_amount`, I think this could be done in master

Reply:
> I think this could be done in master

I agree, but shouldn't this fix be done in 13.0?

Reply:
Sorry, which fix?

Reply:
This PR should be retargetted. It is not because the exact same bug detected in this version cannot be reproduced that way that the bug is not existing.
I'm not willing to test every single template to find one that is not working, but the bug is definitely there.

Reply:
Done: https://github.com/odoo/odoo/pull/90191